source/codegen/metadata/niscope/enums.py

```diff
@@ -695,217 +579,130 @@
-                },
                 'name': 'NISCOPE_VAL_48_TAP_STANDARD',
-                'python_name': 'FOURTYEIGHT_TAP_STANDARD',
                 'value': 0
             },
             {
-                'documentation': {
-                    'description': 'This filter is optimized for the lowest possible bandwidth for a 48 tap filter and maximizes the SNR'
-                },
                 'name': 'NISCOPE_VAL_48_TAP_HANNING',
-                'python_name': 'FOURTYEIGHT_TAP_HANNING',
                 'value': 1
             },
             {
-                'documentation': {
-                    'description': 'This filter is optimized for the lowest possible bandwidth for a 16 tap filter and maximizes the SNR'
-                },
                 'name': 'NISCOPE_VAL_16_TAP_HANNING',
-                'python_name': 'SIXTEEN_TAP_HANNING',
                 'value': 2
             },
             {
-                'documentation': {
-                    'description': 'This filter is optimized for the lowest possible bandwidth for a 8 tap filter and maximizes the SNR'
-                },
                 'name': 'NISCOPE_VAL_8_TAP_HANNING',
-                'python_name': 'EIGHT_TAP_HANNING',
                 'value': 3
             }
         ]
     },
     'GlitchCondition': {
+        'codegen_method': 'public',
         'values': [
             {
-                'documentation': {
-                    'description': 'Trigger on pulses with a duration greater than the specified glitch width.'
-                },
                 'name': 'NISCOPE_VAL_GLITCH_GREATER_THAN',
                 'value': 2
             },
             {
-                'documentation': {
-                    'description': 'Trigger on pulses with a duration shorter than the specified glitch width.'
-                },
                 'name': 'NISCOPE_VAL_GLITCH_LESS_THAN',
                 'value': 1
             }
         ]
     },
     'GlitchPolarity': {
+        'codegen_method': 'public',
         'values': [
             {
-                'documentation': {
-                    'description': 'Trigger on pulses of positive polarity relative to the trigger threshold.'
-                },
                 'name': 'NISCOPE_VAL_GLITCH_POSITIVE',
                 'value': 1
             },
             {
-                'documentation': {
-                    'description': 'Trigger on pulses of negative polarity relative to the trigger threshold.'
-                },
                 'name': 'NISCOPE_VAL_GLITCH_NEGATIVE',
                 'value': 2
             },
             {
-                'documentation': {
-                    'description': 'Trigger on pulses of either positive or negative polarity.'
-                },
                 'name': 'NISCOPE_VAL_GLITCH_EITHER',
                 'value': 3
             }
         ]
     },
     'InputImpedance': {
-        'generate-mappings': True,
```

Comment:
Darn, I was assuming that we always had manually changed doubles to ints when possible for enums in grpc-device. Doesn't seem to be the case though. I can look into updating the hapigen-plugin to take in an option for enums to not try to coerce enum values to ints.

Reply:
I've already made some changes to the hapigen-plugin for Scope specific stuff so I can try updating this as well if you want to have a chat about what all needs to change with me

Reply:
This is being handled in this [PR](https://dev.azure.com/ni/DevCentral/_git/ni-central/pullrequest/366124)

Reply:
This has been fixed